src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1169,241 @@ const isMobileFieldUpdateAllowed = (
     },
   )
 }
+
+export const createTwilioCredentials = (
+  twilioCredentials: TwilioCredentials,
+  formId: string,
+): ResultAsync<unknown, ApplicationError> => {
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+  const msgSrvcName = `formsg/${config.secretEnv}/form/${formId}/twilio`
+  const body: CreateSecretRequest = {
+    Name: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  logger.info({
+    message: `no msgSrvcName, creating Twilio credentials for form ${formId}`,
+    meta: {
+      action: 'createTwilioCredentials',
+      formId: formId,
+      msgSrvcName,
+    },
+  })
+
+  return ResultAsync.fromPromise(
+    FormModel.startSession().then((session: ClientSession) =>
+      session
+        .withTransaction(
+          async () =>
+            await createTwilioTransaction(formId, msgSrvcName, body, session),
```

Comment:
from what I see from the docs, you don't need to pass in a `session` when saving the document as it will use the bounded session, making the `createTwilioTransaction` function less dependent on whether there is a session or not

https://mongoosejs.com/docs/transactions.html#with-mongoose-documents-and-save